sdk/python/feast/feature_store.py

```diff
@@ -296,24 +336,39 @@ def get_historical_features(
             >>> fs = FeatureStore(config=RepoConfig(provider="gcp"))
             >>> retrieval_job = fs.get_historical_features(
             >>>     entity_df="SELECT event_timestamp, order_id, customer_id from gcp_project.my_ds.customer_orders",
-            >>>     feature_refs=["customer:age", "customer:avg_orders_1d", "customer:avg_orders_7d"],
-            >>>     )
+            >>>     features=["customer:age", "customer:avg_orders_1d", "customer:avg_orders_7d"]
+            >>> )
             >>> feature_data = retrieval_job.to_df()
             >>> model.fit(feature_data) # insert your modeling framework here.
         """
-        all_feature_views = self._registry.list_feature_views(project=self.project)
+        _features = features or feature_refs
+        if not _features:
+            raise ValueError("No features specified for retrieval")
```

Comment:
it seems like this whole code block is duplicated, and also different. This block is missing
```
        if isinstance(_features, FeatureService):
            # Get the latest value of the feature service, in case the object passed in has been updated underneath us.
            _features = self.get_feature_service(_features.name)
```
Should we generalize by moving this code into a function?